src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+abstract class AbstractAdapter implements CacheItemPoolInterface
+{
+    private $namespace;
+    private $deferred = array();
+    private $createCacheItem;
+    private $mergeByLifetime;
+
+    protected function __construct($namespace = '', $defaultLifetime = 0)
+    {
+        $this->namespace = $namespace;
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit) use ($defaultLifetime) {
```

Comment:
What about using adding a constructor to `CacheItem` marked with `@internal`? It will be easier to understand. (Currently even PHPStorm isn't able to interpret this snippet and marks it in red).